Controllers/ImageController.cs

```diff
@@ -0,0 +1,336 @@
+                if (new Regex(type).IsMatch("jpeg") || new Regex(type).IsMatch("jpg"))
+                    return new JpegEncoder();
+                if (new Regex(type).IsMatch("webp"))
+                    return new WebpEncoder();
+                if (new Regex(type).IsMatch("bmp"))
+                    return new BmpEncoder();
+                if (new Regex(type).IsMatch("tiff"))
+                    return new TiffEncoder();
+            }
+
+            return new PngEncoder();
+        }
+
+        private static MemoryStream CreateImageStream(ImageGenerationModel model, Profile profile)
+        {
+            // We the individual image has padding specified, used that.
+            // Otherwise, use the general padding passed into the model.
+            var padding = profile.Padding ?? model.Padding;
+
+            var imageEncoder = getEncoderFromType(profile.Format);
+
+            if (model.SvgFormData != null)
+            {
+                return ImageGenerationModel.ProcessSvgToStream(model.SvgFormData, profile.Width, profile.Height, imageEncoder, padding, model.BackgroundColor);
+            }
+            else if (model.BaseImage != null)
+            {
+                return  ImageGenerationModel.ProcessImageToStream(model.BaseImage, profile.Width, profile.Height, imageEncoder, padding, model.BackgroundColor);
+            }
+
+            return null;
+        }
+
+        private static string CreateBase64Image(ImageGenerationModel model, Profile profile)
+        {
+            var formatOrPng = string.IsNullOrEmpty(profile.Format) ? "image/png" : profile.Format;
+            using (var imgStream = CreateImageStream(model, profile))
+            {
+                var base64 = Convert.ToBase64String(imgStream.ToArray());
+                return $"data:{formatOrPng};base64,{base64}";
+            }
+        }
+    }
+
+    public class ImageResponse
+    {
+        public Uri Uri { get; set; }
+    }
+
+
+    public class JsonResponse
+    {
+        public string Json { get; set; }
+    }
+
+    public class IconObject
+    {
+        public IconObject(string src, string size)
+        {
+            this.src = src;
+            this.sizes = size;
+        }
+
+        public string src { get; set; }
+
+        public string sizes { get; set; }
+    }
+
+    public class IconRootObject
+    {
+        public List<IconObject> icons { get; set; } = new List<IconObject>();
+    }
+
+    public class ImageFormData
+    {
+        [Required]
+        public IFormFile fileName { get; set; }
+
+        [Required]
+        [DefaultValue(Platform.android)]
+        public Platform platform { get; set; } /* "android" | "chrome "| "firefox" | "ios" | "msteams" | "windows10" | "windows11" */
+        public string? padding { get; set; }
+
+        public string? color { get; set; }
+    }
+
+    public enum Platform {
+        android,
+        chrome, firefox, ios, msteams, windows10, windows11 }
+}
```

Comment:
Usually you want other classes in their own files.

Typically with C# style you want properties to start with Capital letters. Json converter should map from json object/values.

Reply:
Fixed